wgengine/magicsock/magicsock.go

```diff
@@ -4411,14 +4411,18 @@ func (de *endpoint) addrForWireGuardSendLocked(now mono.Time) (udpAddr netip.Add
 
 	candidates := make([]netip.AddrPort, 0, len(de.endpointState))
 	for ipp := range de.endpointState {
-		if ipp.Addr().Is4() && de.c.noV4.Load() {
```

Comment:
This might be fixing part of an issue right now, but I think it's fixing a symptom of a different problem.

There are several things at play:
- the comment over nov4/nov6 says they're only for logging and only at startup. If they're only for logging we shouldn't be using them here.
- the log data we've inspected suggests neither of the nov* fields should be set, so there's some other bug (perhaps conn.mu is being held too long at startup preventing the netcheck from applying in a timely fashion).
- per brads comment it is possible for a user to get stuck on a particular protocol version here, but it could happen in both directions, stuck on v4 but needing v6 or vice versa. I'm not sure we should be treating either of v4 or v6 specially here, so I'd prefer to see us remove both or neither of these checks.